src/Components/Section.tsx

```diff
@@ -0,0 +1,186 @@
+import {
+  FormEvent, useRef, useState, useEffect,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  onSelect: (value: Todo) => Promise<void>;
+  filteredTodos: Todo[] | null;
+  onDelete: (todoId: number) => void;
+  tempTodo: Todo | null,
+  selectedTodo: Todo | null,
+  isLoading: boolean,
+  loadingTodos: Todo[] | null,
+  onDubleClick: (value: Todo) => void,
+  todos: Todo[],
+};
+
+export const Section: React.FC<Props> = ({
+  onSelect,
+  onDelete,
+  onDubleClick,
+  filteredTodos,
+  tempTodo,
+  isLoading,
+  loadingTodos,
+  selectedTodo,
+}) => {
+  const [currentTodo, setCurrentTodo] = useState<Todo | null>(null);
+  const [updatedQuery, setUpdatedQuery] = useState('');
+  const myInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (currentTodo && myInputRef) {
+      myInputRef.current?.focus();
+    }
+  }, [currentTodo]);
+
+  const handleEscape = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setCurrentTodo(null);
+    }
+  };
+
+  document.addEventListener('keydown', handleEscape);
```

Comment:
better to move it in useEffect, and don't forget to remove the listener on the component unmount